acq_stat_reports/acq_stat_reports.py

```diff
@@ -401,56 +406,61 @@ def acq_fails(acqs, range_datestart, range_datestop, outdir="out"):  #  noqa: AR
     little ones (by mag bin).
     """
 
-    fails = []
+    fails = {}
     range_acqs = acqs[
         (acqs["tstart"] >= range_datestart.cxcsec)
         & (acqs["tstart"] < range_datestop.cxcsec)
     ]
 
-    all_fails = range_acqs[range_acqs["acqid"] == 0]
-    failed_stars = []
-    for fail in all_fails:
-        star = {
-            "id": fail["agasc_id"],
-            "obsid": fail["obsid"],
-            "mag_exp": fail["mag"],
-            "mag_obs": fail["mag_obs"],
+    failed_stars = [
+        {
+            "id": int(mfail["agasc_id"]),
+            "obsid": int(mfail["obsid"]),
+            "mag_exp": mfail["mag"],
+            "mag_obs": mfail["mag_obs"],
             "acq_stat": "NOID",
         }
-        failed_stars.append(star)
-    make_fail_html(failed_stars, outdir / "failed_acq_stars_list.html")
+        for mfail in range_acqs[range_acqs["acqid"] == 0]
+    ]
+    n_stars = len(range_acqs)
+    n_failed = len(np.flatnonzero(range_acqs["acqid"] == 0))
+    fails[""] = {
+        # "mag_range": (-np.inf, np.inf),
+        "n_stars": n_stars,
+        "n_failed": n_failed,
+        "fail_rate": 0 if n_stars == 0 else n_failed / n_stars,
+        "label": "",
+        "failed_stars": failed_stars,
+    }
 
     bin = 0.1
     for tmag_start in np.arange(10.0, 10.8, 0.1):
+        label = f"{tmag_start:.1f}-{tmag_start + bin:.1f}"
         mag_range_acqs = range_acqs[
             (range_acqs["mag"] >= tmag_start) & (range_acqs["mag"] < (tmag_start + bin))
         ]
-        mfailed_stars = []
 
-        mag_fail_acqs = mag_range_acqs[mag_range_acqs["acqid"] == 0]
-        for mfail in mag_fail_acqs:
-            star = {
-                "id": mfail["agasc_id"],
-                "obsid": mfail["obsid"],
+        mfailed_stars = [
+            {
+                "id": int(mfail["agasc_id"]),
+                "obsid": int(mfail["obsid"]),
                 "mag_exp": mfail["mag"],
                 "mag_obs": mfail["mag_obs"],
                 "acq_stat": "NOID",
             }
-            mfailed_stars.append(star)
-        failed_star_file = outdir / f"failed_acq_{tmag_start:.1f}_stars_list.html"
-        make_fail_html(mfailed_stars, failed_star_file)
+            for mfail in mag_range_acqs[mag_range_acqs["acqid"] == 0]
```

Comment:
I find the multi-line list comprehensions a little hard to read but suppose they are efficient.

Reply:
This comment now is outdated. Anyway, at the time, I looked at the code and thought that this particular case was not a difficult one. Copying the old code here:
```python
        mfailed_stars = [
            {
                "id": int(mfail["agasc_id"]),
                "obsid": int(mfail["obsid"]),
                "mag_exp": mfail["mag"],
                "mag_obs": mfail["mag_obs"],
                "acq_stat": "NOID",
            }
            for mfail in mag_range_acqs[mag_range_acqs["acqid"] == 0]
        ]
```

This one is a case about getting used to it. I definitely prefer it.